### PR TITLE
Swift: add `PrintAst`

### PR DESCRIPTION
**Pull request by redsun82**
This includes integration with VSCode.

### CODE REVIEW

**Comment by MathiasVP** [comment]
> We should maybe consider either removing `base` from the schema and add it as a shortcut on QL, or finish up replacing all `SelfApplyExpr` instances with synthesized types.

FWIW, I'm definitely in favor of replacing all `SelfApplyExpr`s with synthesized IPA branches.

File: swift/ql/lib/codeql/swift/printast/PrintAst.qll [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+private predicate shouldPrint(Locatable e) { any(PrintAstConfiguration config).shouldPrint(e) }
+
+/**
+ * An AST node that should be printed.
+ */
+private newtype TPrintAstNode =
+  TLocatable(Locatable ast) {
+    // Only consider resolved nodes (that is not within the hidden conversion AST)
+    ast = ast.resolve()
```
Comment (MathiasVP) [inline swift/ql/lib/codeql/swift/printast/PrintAst.qll:35]: Hmm. Shouldn't this be:
```ql
ast = any(Locatable loc).resolve()
```
?

That is, I think _any_ element targeted by a call to `resolve` should be a possible PrintAST node.

**Review by MathiasVP — COMMENTED** [review]
One question. LGTM otherwise.

File: swift/ql/lib/codeql/swift/printast/PrintAst.qll (hunk 1 above)
Comment (redsun82) [reply swift/ql/lib/codeql/swift/printast/PrintAst.qll:35]: Hmm, I think they are equivalent, because resolve is idempotent.

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/printast/PrintAst.qll]: Ah, right!

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**